src/humanPose/HumanPoseAnalyzer.js

```diff
@@ -0,0 +1,1129 @@
+        this.lenses.forEach(lens => {
+            this.historyLines[lens.name] = {
+                all: {},
+                historical: {},
+                live: {}
+            };
+            this.historyLineContainers.historical[lens.name] = new THREE.Group();
+            this.historyLineContainers.historical[lens.name].visible = lens === this.activeLens;
+            this.historicalHistoryLineContainer.add(this.historyLineContainers.historical[lens.name]);
+            this.historyLineContainers.live[lens.name] = new THREE.Group();
+            this.historyLineContainers.live[lens.name].visible = lens === this.activeLens;
+            this.liveHistoryLineContainer.add(this.historyLineContainers.live[lens.name]);
+        });
+
+        this.clones = {
+            all: [],
+            historical: [],
+            live: []
+        }; // Array of all clones, entry format: Object3Ds with a pose child
+        this.recordingClones = realityEditor.device.environment.isDesktop();
+        this.lastDisplayedClones = [];
+
+        this.prevAnimationState = null;
+        this.animationStart = -1;
+        this.animationEnd = -1;
+        this.animationPosition = -1;
+        this.animationMode = AnimationMode.region;
+        this.lastAnimationTime = Date.now();
+
+        // The renderer for poses that need to be rendered opaquely
+        this.opaquePoseRenderer = new HumanPoseRenderer(new THREE.MeshBasicMaterial(), MAX_POSE_INSTANCES);
+        this.opaquePoseRenderer.addToScene(this.opaqueContainer);
+
+        // Keeps track of the HumanPoseRenderInstances for the start and end of the current selection
+        this.selectionMarkPoseRenderInstances = {
+            start: new HumanPoseRenderInstance(this.opaquePoseRenderer, 'selectionMarkStart', this.activeLens),
+            end: new HumanPoseRenderInstance(this.opaquePoseRenderer, 'selectionMarkEnd', this.activeLens),
+        };
+
+        // Contains all historical poses
+        this.historicalPoseRenderers = [];
+        if (realityEditor.device.environment.isDesktop()) {
+            this.addHistoricalPoseRenderer();
+        }
+
+        // Contains all live-recorded poses
+        this.livePoseRenderers = [];
+        this.addLivePoseRenderer();
+
+        if (realityEditor.device.environment.isDesktop()) {
+            this.settingsUi = new HumanPoseAnalyzerSettingsUi(this);
+            this.setUiDefaults();
+        }
+
+        this.update = this.update.bind(this);
+        window.requestAnimationFrame(this.update);
+    }
+
+    get activeLens() {
+        return this.lenses[this.activeLensIndex];
+    }
+
+    /**
+     * Sets up the containers for the history lines and clones
+     * @param {Object3D} parent - object to add the analyzer's containers to
+     */
+    setupContainers(parent) {
+        this.historicalHistoryLineContainer = new THREE.Group();
```

Comment:
Whenever I see historicalHistory it makes me smile. How about playback instead? For example, historicalHistoryLineContainer  -> playbackHistoryLineContainer  (Its counterpart being liveHistoryLineContainer)

Reply:
I've felt the same way and like this suggestion. Makes it much easier to understand

Reply:
I agree that playback is a better name but renaming this variable would kick off a cascade of historical -> playback that I leave as an exercise for the future.

Reply:
I can tackle that after Liveworx